test/e2e/antreapolicy_test.go

```diff
@@ -311,12 +311,72 @@ func testCNPDropEgress(t *testing.T) {
 	executeTests(t, testCase)
 }
 
+// testBaselineNamespaceIsolation tests that a CNP in the baseline Tier is able to enforce default namespace isolation,
+// which can be later overridden by developer K8s NetworkPolicies.
+func testCNPBaselineNamespaceIsolation(t *testing.T) {
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	nsExpOtherThanX := metav1.LabelSelectorRequirement{
+		Key:      "ns",
+		Operator: metav1.LabelSelectorOpNotIn,
+		Values:   []string{"x"},
+	}
+	builder = builder.SetName("cnp-baseline-isolate-ns-x").
+		SetTier("baseline").
+		SetPriority(1.0).
+		SetAppliedToGroup(nil, map[string]string{"ns": "x"}, nil, nil)
+	builder.AddIngress(v1.ProtocolTCP, &p80, nil, nil, nil, nil,
+		nil, &[]metav1.LabelSelectorRequirement{nsExpOtherThanX}, secv1alpha1.RuleActionDrop)
+
+	// create a K8s NetworkPolicy to allow ingress from pods in it's own ns and y/a to connect to pods in namespace x.
+	k8sNPBuilder := &NetworkPolicySpecBuilder{}
+	k8sNPBuilder = k8sNPBuilder.SetName("x", "default-deny-namespace").
+		SetTypeIngress().
+		AddIngress(v1.ProtocolTCP, &p80, nil, nil, nil,
+			nil, map[string]string{"ns": "x"}, nil, nil).
+		AddIngress(v1.ProtocolTCP, &p80, nil, nil, nil,
+			map[string]string{"pod": "a"}, map[string]string{"ns": "y"}, nil, nil)
+
+	reachability := NewReachability(allPods, true)
+	reachability.Expect(Pod("y/b"), Pod("x/a"), false)
+	reachability.Expect(Pod("y/c"), Pod("x/a"), false)
+	reachability.Expect(Pod("z/a"), Pod("x/a"), false)
+	reachability.Expect(Pod("z/b"), Pod("x/a"), false)
+	reachability.Expect(Pod("z/c"), Pod("x/a"), false)
+	reachability.Expect(Pod("y/b"), Pod("x/b"), false)
+	reachability.Expect(Pod("y/c"), Pod("x/b"), false)
+	reachability.Expect(Pod("z/a"), Pod("x/b"), false)
+	reachability.Expect(Pod("z/b"), Pod("x/b"), false)
+	reachability.Expect(Pod("z/c"), Pod("x/b"), false)
+	reachability.Expect(Pod("y/b"), Pod("x/c"), false)
+	reachability.Expect(Pod("y/c"), Pod("x/c"), false)
+	reachability.Expect(Pod("z/a"), Pod("x/c"), false)
+	reachability.Expect(Pod("z/b"), Pod("x/c"), false)
+	reachability.Expect(Pod("z/c"), Pod("x/c"), false)
```

Comment:
for a future PR, we should find a less verbose way to express this :)

For example:

```go
reachability.Expect(Pod("y/b"), AnyPodInNamespace("x"), false)
reachability.Expect(Pod("y/c"), AnyPodInNamespace("x"), false)
reachability.Expect(Pod("z/a"), AnyPodInNamespace("x"), false)
reachability.Expect(Pod("z/b"), AnyPodInNamespace("x"), false)
reachability.Expect(Pod("z/c"), AnyPodInNamespace("x"), false)
```

Reply:
Added to my TODO list :) I can open an issue for this as well if needed.